src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.targets

```diff
@@ -62,7 +62,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- The CSharp.props used to hard-code WarningLevel to 4, so to maintain parity with .NET Framework projects we do that here. -->
     <WarningLevel Condition="'$(Language)' == 'C#' And '$(WarningLevel)' == '' And '$(TargetFrameworkIdentifier)' == '.NETFramework' ">4</WarningLevel>
     <!-- .NET projects, however, can float up to their TFM's major version -->
-    <WarningLevel Condition="'$(Language)' == 'C#' And '$(WarningLevel)' == '' And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' ">$(_TargetFrameworkVersionWithoutV.Substring(0, 1))</WarningLevel>
+    <WarningLevel Condition="'$(Language)' == 'C#' And '$(WarningLevel)' == '' And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' ">$([System.Version]::Parse('$(_TargetFrameworkVersionWithoutV)').Major</WarningLevel>
```

Comment:
NOT TESTED YET

Reply:
Mind taking a look at https://github.com/dotnet/sdk/blob/main/test%2FMicrosoft.NET.Build.Tests%2FGivenThatWeWantToFloatWarningLevels.cs and seeing if there's a place to add a regression test?

Reply:
Manually tested now, and updated the existing test as well. Let's see if all will be green.